tdrs-backend/tdpservice/parsers/case_consistency_validator.py

```diff
@@ -77,49 +121,185 @@ def validate(self):
-        num_errors += self.__validate_tanf_s1_header_with_records()
+        num_errors += self.__validate_header_with_records()
+        num_errors += self.__validate_s1_records_are_related()
         return num_errors
 
-    def __validate_tanf_s2_case(self, num_errors):
+    def __validate_section2(self, num_errors):
         """Perform TANF Section 2 category four validation on all cached records."""
-        return 0
-
-    def __validate_tribal_tanf_s1_case(self, num_errors):
-        """Perform Tribal TANF Section 1 category four validation on all cached records."""
-        return 0
-
-    def __validate_tribal_tanf_s2_case(self, num_errors):
-        """Perform Tribal TANF Section 2 category four validation on all cached records."""
-        return 0
-
-    def __validate_ssp_s1_case(self, num_errors):
-        """Perform SSP Section 1 category four validation on all cached records."""
-        return 0
-
-    def __validate_ssp_s2_case(self, num_errors):
-        """Perform SSP Section 2 category four validation on all cached records."""
-        return 0
+        num_errors += self.__validate_header_with_records()
+        num_errors += self.__validate_s2_records_are_related()
+        return num_errors
 
-    def __validate_tanf_s1_header_with_records(self):
+    def __validate_header_with_records(self):
         """Header YEAR + header QUARTER must be consistent with RPT_MONTH_YEAR for all T1, T2, and T3 records."""
         year = self.header["year"]
         quarter = self.header["quarter"]
         header_rpt_month_year_list = get_rpt_month_year_list(year, quarter)
         num_errors = 0
-        for record, schema in self.record_schema_pairs:
+        for record, schema in self.record_schema_pairs.cases:
             if record.RPT_MONTH_YEAR not in header_rpt_month_year_list:
                 num_errors += 1
                 err_msg = (f"Failed to validate record with CASE_NUMBER={record.CASE_NUMBER} and "
                            f"RPT_MONTH_YEAR={record.RPT_MONTH_YEAR} against header. If YEAR={year} and "
                            f"QUARTER={quarter}, then RPT_MONTH_YEAR must be in {header_rpt_month_year_list}.")
-                self.generated_errors.append(
-                    self.generate_error(schema=schema,
-                                        error_category=ParserErrorCategoryChoices.CASE_CONSISTENCY,
-                                        error_message=err_msg,
-                                        record=record,
-                                        field="RPT_MONTH_YEAR"
-                                        )
-                                    )
+                self.__generate_and_add_error(
+                    schema,
+                    record,
+                    field='RPT_MONTH_YEAR',
+                    msg=err_msg
+                )
+
+        return num_errors
+
+    def __validate_family_affiliation(self, num_errors, t1s, t2s, t3s, error_msg):
```

Comment:
tests are missing since it has a lot of failure paths

Reply:
both pass and fail are tested in `test_case_consistency`, for all program types. are there more cases you would suggest to cover?